tests/import_export_tests.py

```diff
@@ -554,6 +585,169 @@ def test_labeled_datasets_with_no_labels(basedir, img):
+            basedir, "custom-imports", dataset_type.__name__
+        )
+        _run_custom_imports(udataset, export_dir, dataset_type, max_samples=3)
+
+
+def test_custom_classification_dataset_imports(basedir):
+    # Types of classification datasets to test
+    dataset_types = [
+        fo.types.FiftyOneImageClassificationDataset,
+        fo.types.ImageClassificationDirectoryTree,
+        fo.types.TFImageClassificationDataset,
+    ]
+
+    # Load a small classification dataset
+    cdataset = foz.load_zoo_dataset(
+        "cifar10",
+        split="test",
+        dataset_name="classification-dataset",
+        shuffle=True,
+        max_samples=100,
+    )
+
+    # Remove labeles from some samples
+    for s in cdataset.take(10):
+        s.ground_truth = None
+        s.save()
+
+    # Test custom imports
+    for dataset_type in dataset_types:
+        print(dataset_type.__name__)
+        export_dir = os.path.join(
+            basedir, "custom-imports", dataset_type.__name__
+        )
+        _run_custom_imports(
+            cdataset, export_dir, dataset_type, num_unlabeled=10, max_samples=3
+        )
+
+
+def test_custom_detection_dataset_imports(basedir):
+    # Types of detection datasets to test
+    dataset_types = [
+        fo.types.FiftyOneImageDetectionDataset,
+        fo.types.COCODetectionDataset,
+        fo.types.VOCDetectionDataset,
+        fo.types.KITTIDetectionDataset,
+        fo.types.TFObjectDetectionDataset,
+        fo.types.CVATImageDataset,
+    ]
+
+    # Load a small detection dataset
+    ddataset = foz.load_zoo_dataset(
+        "coco-2017",
+        split="validation",
+        dataset_name="detection-dataset",
+        shuffle=True,
+        max_samples=100,
+    )
+
+    # Remove labeles from some samples
+    for s in ddataset.take(10):
+        s.ground_truth = None
+        s.save()
+
+    # Test custom imports
+    for dataset_type in dataset_types:
+        print(dataset_type.__name__)
+
+        # COCODetectionDataset and TFObjectDetectionDataset formats cannot
+        # distinguish between an unlabeled image and a labeled image with zero
+        # detections
+        num_unlabeled = 10
+        if dataset_type in (
+            fo.types.COCODetectionDataset,
+            fo.types.TFObjectDetectionDataset,
+        ):
+            num_unlabeled = None
+
+        export_dir = os.path.join(
+            basedir, "custom-imports", dataset_type.__name__
+        )
+        _run_custom_imports(
+            ddataset,
+            export_dir,
+            dataset_type,
+            num_unlabeled=num_unlabeled,
+            max_samples=3,
+        )
+
+
+def test_custom_multitask_image_dataset_imports(basedir):
+    # Types of multitask datasets to test
+    dataset_types = [
+        fo.types.FiftyOneImageLabelsDataset,
+        fo.types.BDDDataset,
+    ]
+
+    # Load a small multitask image dataset
+    idataset = foz.load_zoo_dataset(
+        "coco-2017",
```

Comment:
A couple problems this is causing with tests: `pycocotools` is required now (which can be fixed), and this dataset is at least a 760 MB download (6.1 Gb according to the progress bar), so it will make tests take a lot longer to run. Is there a smaller dataset that these tests could be run on? Alternatively, could these tests be moved to a subfolder that is skipped by CI? Currently, `tests/benchmarking/` is skipped, but we could create a new folder if needed.

Reply:
Made a branch to test this: https://github.com/voxel51/fiftyone/runs/1155486529?check_suite_focus=true
`import_export_tests` took maybe 2 minutes longer than usual, which isn't that bad, but it would still be nice to speed it up.

Reply:
I like the idea of moving `import_export_tests.py` and `zoo_tests.py` to a new subdirectory that is skipped by CI.

I'd rather not go through the work required to write small, non-zoo datasets to test with right now. Import/export is inherently data intensive, and we'll probably want to add beefier tests at some point anyway. Basically, I'm fine with having to manually run tests locally related to import/export functionality.

Also, the `zoo_tests.py` (which are currently very weak) would ideally be upgraded someday to test all zoo datasets, at which point there will inevitably be lots of I/O and the tests should not be run as often as other unit tests.

Reply:
Can you move `import_export_tests.py` and `zoo_tests.py` to a new subdirectory and omit from CI?

Reply:
Sure, I can do that on this branch if you like.
Initial results from my experimental test branch: Linux tests take 2-3 minutes longer to run, and Windows tests haven't finished yet (so they would be at least 5-6 minutes slower). Agreed that if we plan to expand these tests, they should not normally be run.

Reply:
a4d80c50
feel free to rename as long as you also update `build.yml`